### PR TITLE
feat: support private projects for event search

### PR DESCRIPTION
**Pull request by sjaanus**
Adds private projects support for event search. Unit test should cover the usecases.

### CODE REVIEW

File: src/lib/features/events/event-service.test.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+        const result = filterPrivateProjectsFromParams(
+            projectParam,
+            projectAccess,
+        );
+
+        expect(result).toBe('IS:project1');
+    });
+
+    it('should throw an error if no projects match and not throw an error', () => {
```
Comment (thomasheartman) [inline src/lib/features/events/event-service.test.ts:56]: Huh? What's this test supposed to say?  Something like this?
```suggestion
    it('should throw an error if no projects match', () => {
```

File: src/lib/features/events/event-service.test.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+        expect(result).toBe('IS_ANY_OF:project1,project2');
+    });
+
+    it('should return the original projectParam when mode is all', () => {
+        const projectAccess: ProjectAccess = {
+            mode: 'all',
+        };
+
+        const projectParam = 'IS:project1,IS:project3';
```
Comment (thomasheartman) [inline src/lib/features/events/event-service.test.ts:23]: Project param the query parameter in the URL, right? If you have multiple projects, it's formatted as an `IS_ANY_OF:`, not as a concatenated list of `IS` (e.g. https://sandbox.getunleash.io/enterprise/history?query=&limit=25&offset=0&project=IS_ANY_OF%3A61ab6557353d%2Candreas_test).

We should probably use the same kinda query string, yeah?

(Side note: how does the API handle multiples of the same query param? E.g. "project=IS_ANY_OF:a,b&project=IS:c"?)

**Review by thomasheartman — APPROVED** [review]

File: src/lib/features/events/event-service.test.ts (hunk 1 above)
Comment (sjaanus) [reply src/lib/features/events/event-service.test.ts]: Yeah, fixed

File: src/lib/features/events/event-service.test.ts (hunk 2 above)
Comment (sjaanus) [reply src/lib/features/events/event-service.test.ts]: Fixed. Side note, not sure actually, but it is not supposed to be used that way, and only doable via API.

**Pull request merged**